src/Core/Localization/Specification/Factory.php

```diff
@@ -0,0 +1,277 @@
+            $this->getPositivePattern($currencyPattern),
+            $this->getNegativePattern($currencyPattern),
+            $this->computeNumberSymbolLists($numbersSymbols),
+            $precision,
+            $this->getMinFractionDigits($currencyPattern),
+            $numberGroupingUsed,
+            $this->getPrimaryGroupSize($currencyPattern),
+            $this->getSecondaryGroupSize($currencyPattern),
+            $currencyDisplayType,
+            $currency->getSymbol($localeCode),
+            $currency->getIsoCode()
+        );
+    }
+
+    /**
+     * Extract the positive pattern from a CLDR formatting pattern
+     * Works with any formatting pattern (number, price, percentage).
+     *
+     * @param string $pattern
+     *                        The CLDR pattern
+     *
+     * @return string
+     *                The extracted positive pattern
+     */
+    protected function getPositivePattern($pattern)
+    {
+        $patterns = explode(';', $pattern);
+
+        return isset($patterns[0]) ? $patterns[0] : '';
+    }
+
+    /**
+     * Extract the negative pattern from a CLDR formatting pattern
+     * Works with any formatting pattern (number, price, percentage).
+     *
+     * @param string $pattern
+     *                        The CLDR pattern
+     *
+     * @return string
+     *                The extracted negative pattern
+     */
+    protected function getNegativePattern($pattern)
+    {
+        $patterns = explode(';', $pattern);
+
+        return isset($patterns[1])
+            ? $patterns[1]
+            : '-' . $patterns[0];
+    }
+
+    /**
+     * Convert a list of CLDR number symbols data into a list of NumberSymbolList objects.
+     *
+     * @param NumberSymbolsData[] $allNumberSymbolsData
+     *                                                  All the CLDR number symbols data indexed by numbering system
+     *
+     * @return NumberSymbolList[]
+     *
+     * @throws LocalizationException
+     *                               If passed data is invalid
+     */
+    protected function computeNumberSymbolLists($allNumberSymbolsData)
+    {
+        $symbolsLists = [];
+        foreach ($allNumberSymbolsData as $numberingSystem => $numberSymbolsData) {
+            $symbolsLists[$numberingSystem] = $this->getNumberSymbolList($numberSymbolsData);
+        }
+
+        return $symbolsLists;
+    }
+
+    /**
+     * Get a NumberSymbolList object from a CLDR NumberSymbolsData object.
+     *
+     * @param NumberSymbolsData $symbolsData
+     *                                       Data that will be used to build the NumberSymbolList object
+     *
+     * @return NumberSymbolList
+     *                          An immutable NumberSymbolList object
+     *
+     * @throws LocalizationException
+     *                               If passed data is invalid
+     */
+    protected function getNumberSymbolList(NumberSymbolsData $symbolsData)
+    {
+        return new NumberSymbolList(
```

Comment:
Too much parameters :(, why not checking for a $symbolData parameter?

Reply:
I'm not sure I understand what you mean @PierreRambaud ?
Do you suggest creating something like `NumberSymbolList ::createFromData(NumberSymbolsData $symbolsData)` ? Or the other way `NumberSymbolsData::createNumberSymbolList()`?

Reply:
this means checking the type of first parameter.. allowing two types for the first symbol and all the others are optional..
not a big fan of this method, I prefer an explicit static function with strong typing `NumberSymbolList ::createFromData(NumberSymbolsData $symbolsData)` so that there is no confusion